### PR TITLE
Accommodate EC pool coverage for snapshot based mirroring test

### PR DESCRIPTION
**Pull request by sunilangadi2**
# Description
Fixes RHCEPHQE-7591
To cover polarion test case https://polarion.engineering.redhat.com/polarion/redirect/project/CEPH/workitem/question_mark/id=CEPH-83573616

Test Flow:
Added module to adapt EC pool coverage
As this test case says to cover snapshot based mirroring along EC pools 
Hence plan is to mark this TC as Inactive by adding step to cover EC pools in TC CEPH-83575375

Test Result:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-O2RQ7X

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/rbd_mirror/test_rbd_mirror_snapshot.py [hunk 1]
```diff
@@ -108,6 +97,53 @@ def run(**kw):
...
+    mirror_obj = rbd_mirror_config(**kw)
+
+    if mirror_obj:
+        log.info("Executing test on replicated pool")
+        if test_snapshot_schedule(
+            mirror_obj.get("rep_rbdmirror"), "rep_pool_config", **kw
+        ):
+            return 1
+
+        log.info("Executing test on ec pool")
```
Comment (rkhambad) [inline tests/rbd_mirror/test_rbd_mirror_snapshot.py:143]: Log did not show any execution on ec pool.

Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_snapshot.py:143]: sorry it ran against wrong branch
will update success logs soon it's been running
do we have any suggestion apart from log?

File: suites/pacific/rbd/tier-2_rbd_mirror_snapshot_regression.yaml [hunk 2]
```diff
@@ -154,8 +154,12 @@ tests:
...
           config:
-            imagesize: 2G
-            io-total: 200M
+            ec_pool_config:
+              mirrormode: snapshot
+              mode: image
+            rep_pool_config:
+              mirrormode: snapshot
+              mode: image
             snapshot_schedule_level: "pool" #snapshots can be scheduled at pool, image or cluster(default) level
```
Comment (VasishtaShastry) [inline suites/pacific/rbd/tier-2_rbd_mirror_snapshot_regression.yaml:163]: [Optional] 
Can we remove this comment?
This looks like a usage info and usage infos can be limited to modules ?

**Review by VasishtaShastry — CHANGES_REQUESTED** [review]
1) Can you please provide apt logs ?
2) Can you please modify TC in description (mark as inactive?) ?

File: tests/rbd_mirror/test_rbd_mirror_snapshot.py (hunk 1 above)
Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_snapshot.py:143]: Attached proper success logs.

File: suites/pacific/rbd/tier-2_rbd_mirror_snapshot_regression.yaml (hunk 2 above)
Comment (sunilangadi2) [reply suites/pacific/rbd/tier-2_rbd_mirror_snapshot_regression.yaml]: Done.

**Comment by sunilangadi2** [comment]
Updated test results and polarion test case.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-O2RQ7X
Please review it.

File: tests/rbd_mirror/test_rbd_mirror_snapshot.py [hunk 3]
```diff
@@ -108,6 +97,53 @@ def run(**kw):
...
+    2. Add mirror snapshot schedule at cluster and pool level
+    3. create some images and wait for them to mirror to secondary
+    4. Check that mirror snapshots are created for each images
+    5. Create a snapshot mirror schedule and list the snapshots
+    6. View the status of schedule snapshot
+    7. remove the snapshot schedule from cluster and pool level, verify the same
+    8. Perform test steps for both Replicated and EC pool
+    """
+    log.info(
+        "Starting CEPH-83575375 and CEPH-83575376, \
```
Comment (rkhambad) [inline tests/rbd_mirror/test_rbd_mirror_snapshot.py:130]: one small change, 
can you please remove this \ , its creating tab (space) in log.

Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_snapshot.py]: Done thanks.

**Review by rkhambad — DISMISSED** [review]
LGTM

File: tests/rbd_mirror/test_rbd_mirror_snapshot.py [hunk 4]
```diff
@@ -1,43 +1,32 @@
...
+    """Method to validate snapshot based mirroring schedule
+    at cluster and pool level.
+    1) Create schedulers at multiple level and verify the same
+    2) Remove schedulers at multiple level and verify the same
+
     Args:
-        **kw:
+        rbd_mirror: Object
+        pool_type: Replicated or EC pool
+        **kw: test data
```
Comment (sunilkumarn417) [inline tests/rbd_mirror/test_rbd_mirror_snapshot.py:16]: please add example config of kw **args

File: tests/rbd_mirror/test_rbd_mirror_snapshot.py [hunk 5]
```diff
@@ -108,6 +97,53 @@ def run(**kw):
...
         log.exception(e)
         return 1
+
+
+def run(**kw):
+    """
+    Snapshot based rbd mirroring schedulers at cluster and pool level.
+
+    Args:
+        **kw: test data
```
Comment (sunilkumarn417) [inline tests/rbd_mirror/test_rbd_mirror_snapshot.py:111]: please add example of **kw data

Comment (sunilangadi2) [reply tests/rbd_mirror/test_rbd_mirror_snapshot.py:134]: Done

**Review by rkhambad — APPROVED** [review]
LGTM

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**